File: src/start.go
package src

import (
	"fmt"
	"songBot/src/config"
	"time"

	"github.com/amarnathcjd/gogram/telegram"
)

// startHandle responds to the /start command with a welcome message.
func startHandle(m *telegram.NewMessage) error {
	bot := m.Client.Me()
	name := m.Sender.FirstName
	go func() {
		err := config.SaveUser(m.Sender.ID)
		if err != nil {
			m.Client.Logger.Error("Save user error:", err)
		}
	}()

	response := fmt.Sprintf(
		`👋 Hello <b>%s</b>!

🎧 <b>Welcome to %s</b> — your personal music downloader bot!  
Supports: <b>Spotify</b>, <b>YouTube</b>, <b>Apple Music</b>, and <b>SoundCloud</b>.

🔍 <b>To search:</b> Send a song name or a link.
💬 <b>Inline Search:</b> <code>@%s lofi mood</code>
📥 <b>Group Command:</b> <code>/spotify &lt;url&gt;</code>

Enjoy endless tunes! 🚀`,
		name,
		bot.FirstName,
		bot.Username,
	)

	keyboard := telegram.NewKeyboard().
		AddRow(telegram.Button.URL("💫 Fᴀʟʟᴇɴ Pʀᴏᴊᴇᴄᴛꜱ", "https://t.me/FallenProjects"))

	_, err := m.Reply(response, telegram.SendOptions{
		ReplyMarkup: keyboard.Build(),
	})
	return err
}

// pingHandle responds to the /ping command with the bot's latency.
func pingHandle(m *telegram.NewMessage) error {
	start := time.Now()

	msg, err := m.Reply("⏱️ Pinging...")
	if err != nil {
		return err
	}

	latency := time.Since(start)
	_, err = msg.Edit(fmt.Sprintf("🏓 <b>Pong!</b> <code>%s</code>", latency))
	return err
}

File: src/modules/start.py
import time
from datetime import datetime

from pytdbot import Client, types

from src import __version__, StartTime
from src.utils import Filter

@Client.on_message(filters=Filter.command(["start", "help"]))
async def start_cmd(c: Client, message: types.Message):
    bot_username = c.me.usernames.editable_username
    text = (
        "<b>🎧 Welcome to SpTube Bot</b>\n\n"
        "Stream, download, and enjoy music from your favorite platforms:\n"
        "• <b>Spotify</b>\n"
        "• <b>YouTube</b>\n"
        "• <b>SoundCloud</b>\n"
        "• <b>Apple Music</b>\n\n"
        "<b>🎥 Now also supports media from:</b>\n"
        "• <b>Instagram</b> (Reels, Posts, Stories)\n"
        "• <b>Pinterest</b>\n"
        "• <b>Facebook</b> (Videos)\n"
        "• <b>TikTok</b>\n\n"
        f"<b>📚 Version:</b> <code>{__version__}</code>\n"
        "📥 <b>How to use:</b>\n"
        "• Send a song name, link, or media URL directly.\n"
        f"• Use inline mode: <code>@{bot_username} your search</code>\n\n"
        "📜 <b>Privacy Policy:</b> /privacy"
    )

    reply = await message.reply_text(
        text,
        parse_mode="html",
        disable_web_page_preview=True,
        reply_markup=types.ReplyMarkupInlineKeyboard(
            [
                [
                    types.InlineKeyboardButton(
                        text="Add me to your group",
                        type=types.InlineKeyboardButtonTypeUrl(
                            f"https://t.me/{bot_username}?startgroup=true"
                        ),
                    ),
                    types.InlineKeyboardButton(
                        text="GitHub",
                        type=types.InlineKeyboardButtonTypeUrl(
                            "https://github.com/AshokShau/SpTubeBot"
                        ),
                    )
                ]
            ]
        ),
    )
    if isinstance(reply, types.Error):
        c.logger.warning(f"Error sending start/help message: {reply.message}")



@Client.on_message(filters=Filter.command("privacy"))
async def privacy_handler(_: Client, message: types.Message):
    await message.reply_text(
        "🔒 <b>Privacy Policy</b>\n\n"
        "This bot does <b>not store</b> any personal data or chat history.\n"
        "All queries are processed in real time and nothing is logged.\n\n"
        "🛠️ <b>Open Source</b> — You can inspect and contribute:\n"
        "<a href=\"https://github.com/AshokShau/SpTubeBot\">github.com/AshokShau/SpTubeBot</a>",
        parse_mode="html",
        disable_web_page_preview=True
    )



@Client.on_message(filters=Filter.command("ping"))
async def ping_cmd(client: Client, message: types.Message) -> None:
    start_time = time.monotonic()
    reply_msg = await message.reply_text("🏓 Pinging...")
    latency = (time.monotonic() - start_time) * 1000  # in ms
    uptime = datetime.now() - StartTime
    uptime_str = str(uptime).split(".")[0]

    response = (
        "📊 <b>System Performance Metrics</b>\n\n"
        f"⏱️ <b>Bot Latency:</b> <code>{latency:.2f} ms</code>\n"
        f"🕒 <b>Uptime:</b> <code>{uptime_str}</code>\n"
        f"🤖 <b>Bot Version:</b> <code>{__version__}</code>"
    )
    done = await reply_msg.edit_text(response, disable_web_page_preview=True)
    if isinstance(done, types.Error):
        client.logger.warning(f"Error sending message: {done}")
    return None

